ref(rust): clarify logs defaults in new release (#15036)
getsentry/sentry-docs: docs/platforms/rust/guides/tracing/index.mdx

}
```

> By default, error level events from `tokio-rs/tracing` are captured as Sentry events, while anything at or above info level is added as a breadcrumb.

Reviewer: l: grammar/stylenits

```suggestion
By default, error-level events from `tokio-rs/tracing` are captured as Sentry events, and events at or above the info level are captured as breadcrumbs.
```

> Additionally, if the `logs` feature flag of the `sentry` crate is enabled and the SDK is initialized with `enable_logs: true`, then events from `tokio-rs/tracing` at info level or above are also captured as [structured logs](https://docs.sentry.io/product/explore/logs/).

Reviewer: [question unrelated to this PR]: Needing to enable both the `logs` feature flag and `enable_logs: true` seems to be like a double-opt-in. Could we make it so that only the feature flag is needed?

Author: Introducing the client option was mostly for the sake of having a uniform API across SDKs, but I agree it would make sense to just use the feature flag in this case, I'll discuss it internally.

Reviewer: As long as the feature is opt-in (not a default feature), I think it makes sense!
